src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -170,26 +182,55 @@ private ProgressStepDefinition[] CreateWorkflowSteps(IProgressController control
             }
         }
 
-        internal /*for testing purposes*/ void SilentSaveSolutionIfDirty()
+        internal /*for testing purposes*/ void DownloadBindingParameters(IProgressController controller, CancellationToken token, IProgressStepExecutionEvents notifications)
         {
-            bool saved = VsShellUtils.SaveSolution(this.host, silent: true);
-            Debug.Assert(saved, "Should not be cancellable");
+            notifications.ProgressChanged(Strings.PreparingBindingWorkflowProgessMessage, double.NaN);
+
+            var properties = this.host.SonarQubeService.GetProperties(token);
+
+            if (token.IsCancellationRequested)
+            {
+                AbortWorkflow(controller, token);
+                return;
+            }
+
+            var testProjRegexProperty = properties.FirstOrDefault(x => StringComparer.Ordinal.Equals(x.Key, ServerProperty.TestProjectRegexKey));
+            var testProjRegexPattern = testProjRegexProperty?.Value ?? ServerProperty.TestProjectRegexDefaultValue;
+
+            this.projectFilter.SetTestRegex(new Regex(testProjRegexPattern, RegexOptions.IgnoreCase));
```

Comment:
This is a potential attack vector - we need to update the threat model. CC: @bgavrilMS 
More info: https://msdn.microsoft.com/en-us/magazine/ff646973.aspx
I remember dealing with similar bug by specifying a timeout to avoid DoS attacks.